File: practice28/1.java


//https://www.nowcoder.com/practice/80ce674313ff43af9d7ac7a41ae21527?tpId=98&tqId=33025&tPage=11&rp=11&ru=/ta/2019test&qru=/ta/2019test/question-ranking
import java.util.Scanner;

public class Main{
    public static void main(String args[]){
    Scanner scanner=new Scanner(System.in);
        String s = scanner.next();
        String str[] = s.split("\\.");
        if (Integer.parseInt(str[0]) > 255 || Integer.parseInt(str[1]) > 255 || Integer.parseInt(str[2]) > 255 || Integer.parseInt(str[3]) > 255) {
            System.out.println(0);
        } else if (Integer.parseInt(str[0]) == 10 && Integer.parseInt(str[1]) >= 0 && Integer.parseInt(str[2]) >= 0 && Integer.parseInt(str[3]) >= 0) {
            System.out.println(1);
        } else if (Integer.parseInt(str[0]) == 172 && Integer.parseInt(str[1]) >= 16 && Integer.parseInt(str[1]) <= 31 && Integer.parseInt(str[2]) >= 0 && Integer.parseInt(str[3]) >= 0) {
            System.out.println(1);
        } else if (Integer.parseInt(str[0]) == 192 && Integer.parseInt(str[1]) == 168 && Integer.parseInt(str[2]) >= 0 && Integer.parseInt(str[3]) >= 0) {
            System.out.println(1);
        }
        else{
            System.out.println(0);
        }
    }
}

File: practice33/1.py

#https://www.nowcoder.com/practice/02d8d42b197646a5bbd0a98785bb3a34?tpId=85&tqId=29857&tPage=2&rp=2&ru=/ta/2017test&qru=/ta/2017test/question-ranking
num=input().split(" ")
n1=int(num[0])
n2=int(num[1])
n3=int(num[2])
n4=int(num[3])
A=(n1+n3)//2
B=(n2+n4)//2
C=(n4-n2)//2
if(A-B==n1 and B-C==n2 and A+B==n3 and B+C==n4):
    print(str(A)+" "+str(B)+" "+str(C))
else:
    print("No")